Feature/31 Unauthorized user leisure page, flow

ivhorodko opened the pull request.

ivhorodko committed:
Remove old map

ivhorodko committed:
Add lacalisation for placeState

ivhorodko committed:
Add more controls for mapSettings

ivhorodko committed:
Remove old path for map

ivhorodko committed:
Update map container and update place cards

ivhorodko committed:
Update leisure status

ivhorodko committed:
Add leisure-map script

ivhorodko committed:
Remove eslint disable

sanioooook commented on front/src/views/leisure/leisure.view.css:
```diff
@@ -62,7 +63,6 @@
 
 .places {
   flex-wrap: wrap;
-  justify-content: start;
   justify-content: space-between;
```
this can be achieved by using classes
https://vuetifyjs.com/ru/styles/flex/#flex-wrap
https://vuetifyjs.com/ru/styles/flex/#flex-justify

ivhorodko replied:
thx

horba commented on front/src/views/leisure/leisure.view.css:
```diff
@@ -122,4 +117,57 @@
+.rating-wrap .star {
+  font-size: 18px;
+  margin-right: 3px;
+  color: #4DD0E1;
```
use variables

horba commented on front/src/views/leisure/leisure.view.css:
```diff
@@ -122,4 +117,57 @@
+  height: 88px;
+  justify-content: center;
+  align-items: center;
+  background: #FFF;
```
we have css variables

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,175 @@ export default {
+      getNextPage: null,
+      google: null,
+      map: null,
+      apiKey: 'AIzaSyDuiiLqO-HTj8vP0POPrsBGJdWc7esHdOY',
```
it is not the best place to store apiKey

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,175 @@ export default {
+      }
+    }
+  },
+  watch: {
+    showMap (value) {
+      if (value) {
+        this.initializeMap();
+      }
+    }
+  },
```
do we realy need watcher? might we can use just computed property or other ways?

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,175 @@ export default {
+    trimString (value) {
+      return value.length > 15 ? value.substr(0, 15) + '...' : value;
+    },
+    opacityPrice (value) {
```
i don't understand this name and what this filter does

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,175 @@ export default {
+        return value;
+      }
+    },
+    nonOpacityPrice (value) {
```
i don't understand this name and what this filter does

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,175 @@ export default {
+        return value;
+      }
+    }
+  },
```
just think if some of this filters are useful for other components

horba replied:
actual

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,175 @@ export default {
+          setTimeout(this.dropMarker(i), i * 100);
+        });
+        this.places = places;
+        console.log(places);
```
atatata

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,175 @@ export default {
+          };
+
+          this.markers[i].placeResult = place;
+          setTimeout(this.dropMarker(i), i * 100);
```
why do we need this timeout?

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,175 @@ export default {
+            }
+          }
+
+          this.getNextPage = pagination.hasNextPage && function () {
```
what is it getNextPage? it seems you said that we don\t know total amount to use pagination

horba reviewed (changes requested):
please, provide separate component to work with google map and encapsulate logic of map.  Just notice that component could have properties and emit custom events to make it more useful, independent and reusable

ivhorodko committed:
add leisure css variables #feature/31

ivhorodko committed:
change leisure styles #feature/31

replace color to css variables. Remove unnecessary styles

ivhorodko committed:
Add leisure place star variables and remove old star color #feature/31

ivhorodko committed:
add rating component

ivhorodko committed:
Resolve pull request problems

1. Raname getNextPage - loadNextPage\n2. Set API Key to secrets\n3. Add dropMarkerAnimation func\n4. Remove price filters

horba commented on front/src/views/leisure/leisure.view.css:
```diff
@@ -122,4 +116,55 @@
+  background:var(--leisure-containers-bgc);
+}
+
+.price-wrap {
+  justify-self: center;
+  align-items: center;
+}
+
+.price-wrap >>> .v-rating--dense .v-icon {
+  padding: 0;
+}
+
+.price-wrap .opacity {
+  opacity: .5;
+}
+
+.price-wrap p {
+  padding: 0px 5px;
+  font-size: 10px;
+  margin:0px;
+}
+
+.rating{
+  justify-content: center;
+  align-items: center;
+}
+
+.price-wrap .state {
+  color: var(--leisure-price-state-color);
+}
+
+.place-info span {
+  font-size: 15px;
+}
+
```
are you sure that it is not useful for other screens?

Commit:
resolve conflict, remove old map #31

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,156 @@ export default {
+        bounds: this.map.getBounds(),
+        types: this.leisureTypes,
+        location: position,
+        radius: this.radius * 1000,
```
why 1000?

horba commented on front/src/views/leisure/leisure.view.js:
```diff
@@ -15,32 +16,156 @@ export default {
-      ]
+      markers: [],
+      places: [],
+      mapCenter: { lat: 48.459286, lng: 35.052698 },
```
why this coords?
